pkg/v10/resource/chart/create.go

```diff
@@ -22,79 +22,88 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 		return microerror.Mask(err)
 	}
 
-	tenantHelmClient, err := r.getTenantHelmClient(ctx, obj)
-	if tenantcluster.IsTimeout(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "did not get a Helm client for the guest cluster")
-
-		// A not found error here means that the cluster-operator certificate for
-		// the current guest cluster was not found. We can't continue without a Helm
-		// client. We will retry during the next execution, when the certificate
-		// might be available.
-		resourcecanceledcontext.SetCanceled(ctx)
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-
-		return nil
-	} else if helmclient.IsTillerInstallationFailed(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "Tiller installation failed")
-
-		// Tiller installation can fail during guest cluster setup. We will retry
-		// on next reconciliation loop.
-		resourcecanceledcontext.SetCanceled(ctx)
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-
-		return nil
-	} else if guest.IsAPINotAvailable(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "Guest API not available.")
-
-		// We should not hammer guest API if it is not available, the guest cluster
-		// might be initializing. We will retry on next reconciliation loop.
-		resourcecanceledcontext.SetCanceled(ctx)
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-
-		return nil
-	} else if err != nil {
-		return microerror.Mask(err)
-	}
-
 	if !reflect.DeepEqual(createState, ResourceState{}) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "creating chart-operator chart")
-
-		tarballPath, err := r.apprClient.PullChartTarball(ctx, createState.ChartName, chartOperatorChannel)
-		if err != nil {
-			return microerror.Mask(err)
+		var tenantHelmClient helmclient.Interface
```

Comment:
* The changes for the current, create, update and delete steps are basically the same. I moved the logic for calling `getTenantHelmClient` into the conditional scope where it is necessary to be used. When computing the Helm client we have to do a lot of network requests to other dependencies and also ensure the Tiller installation all the time. This is not necessary so the situation improves already. 

* I was thinking that we should rather have a dedicated resource for the Tiller installation running before the `chart` resource, but am not sure how much energy we put into it as Tiller is basically on life support with Helm V3. 

* I was thinking that the `chart` resource should be called `chartoperator` because it is installing `chart-operator` in the tenant clusters as I understood it.

Reply:
I think tiller installation/upgrade should happen only during the boot time. One problem is we don't have good error handling during the boot time ATM.

Reply:
Aslo I don't know if it makes sense to invest time into this resource when app-operator is around the corner. release-opreator will install app-operator required for the release.

Reply:
Yes I don't think investing lots of time here is a good idea. This will move to app-operator and yes the plan is to call the resource `chartoperator` because its a much clearer name.

For installing Tiller this is to install it in tenant clusters. So boot time doesn't work. It should happen the first time a tenant cluster becomes accessible.

We also need to be able to upgrade Tiller so it isn't a one time operation. But reducing the number of network calls to achieve this is a good thing.

Reply:
In general when there is something to be installed somewhere and when this process takes time, we should have a dedicated resource for it. The resource should initiate the installation and not block. On the next reconciliation loop (RL) the resource should see how far it is and do more if required. When checking the status or something else is difficult or time consuming for the resource we could think about some status we can write and read to know where we are at. Just some thoughts for the future.

Reply:
Yes unfortunately we'll need to run Tiller for a while longer. This will get reworked when we move the logic to app-operator. So having a specific resource to ensure Tiller is installed makes sense to me. We'll keep it in mind.

cc @giantswarm/team-batman

Reply:
Interesting, why do we need Tiller for longer? Above you said we should not invest much time here. Do you think the time investment into the Tiller resource would be well spent?

Reply:
It's not clear when Helm 3 will be released and production ready. https://github.com/helm/helm/projects/1

There are other options for removing Tiller but I'd prefer to see what Helm 3 comes up with.

>  Do you think the time investment into the Tiller resource would be well spent?

If we still need Tiller when we migrate this code to app-operator then adding a tiller resource likely make sense. My issue is I don't want to invest a lot of time into `cluster-operator`. The 3 controller structure is painful to work with and this goes away with `app-operator`.